e2e/tests/transfer/authz_test.go

```diff
@@ -0,0 +1,142 @@
+package transfer
+
+import (
+	"context"
+	"testing"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	test "github.com/strangelove-ventures/ibctest/v6/testutil"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+)
+
+func TestAuthzTransferTestSuite(t *testing.T) {
+	suite.Run(t, new(AuthzTransferTestSuite))
+}
+
+type AuthzTransferTestSuite struct {
+	testsuite.E2ETestSuite
+}
+
+func (suite *AuthzTransferTestSuite) TestAuthz_MsgTransfer_Succeeds() {
+	t := suite.T()
+	ctx := context.TODO()
+
+	relayer, channelA := suite.SetupChainsRelayerAndChannel(ctx, transferChannelOptions())
+	chainA, chainB := suite.GetChains()
+
+	chainADenom := chainA.Config().Denom
+
+	granterWallet := suite.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	granterAddress := granterWallet.Bech32Address(chainA.Config().Bech32Prefix)
+
+	granteeWallet := suite.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	granteeAddress := granteeWallet.Bech32Address(chainA.Config().Bech32Prefix)
+
+	receiverWallet := suite.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	receiverWalletAddress := receiverWallet.Bech32Address(chainB.Config().Bech32Prefix)
+
+	t.Run("start relayer", func(t *testing.T) {
+		suite.StartRelayer(relayer)
+	})
+
+	t.Run("broadcast MsgGrant", func(t *testing.T) {
+		transferAuth := transfertypes.TransferAuthorization{
+			Allocations: []transfertypes.Allocation{
+				{
+					SourcePort:       channelA.PortID,
+					SourceChannel:    channelA.ChannelID,
+					SpendLimit:       sdk.NewCoins(sdk.NewCoin(chainADenom, sdk.NewInt(testvalues.StartingTokenAmount))),
+					AllowList: []string{receiverWalletAddress},
+				},
+			},
+		}
+
+		authAny, err := codectypes.NewAnyWithValue(&transferAuth)
+		suite.Require().NoError(err)
+
+		msgGrant := &authz.MsgGrant{
+			Granter: granterAddress,
+			Grantee: granteeAddress,
+			Grant: authz.Grant{
+				Authorization: authAny,
+				// no expiration
+				Expiration: nil,
+			},
+		}
+
+		resp, err := suite.BroadcastMessages(context.TODO(), chainA, granterWallet, msgGrant)
+		suite.AssertValidTxResponse(resp)
+		suite.Require().NoError(err)
+	})
+
+	t.Run("exec MsgTransfer", func(t *testing.T) {
```

Comment:
wonder if we should reword this a little 

```suggestion
	t.Run("broadcast MsgExec for ibc MsgTransfer", func(t *testing.T) {
```